Tests/scripts/convert_test_playbook_result_to_jira_issues.py

```diff
@@ -190,16 +186,18 @@ def main():
             if jira_ticket or total_errors:
                 # if the ticket isn't resolved, or we found new errors, we update it, otherwise we skip it.
                 if jira_ticket and jira_ticket.get_field("resolution") and not total_errors:
-                    playbook_to_jira_mapping[playbook_id] = jira_ticket_to_json_data(jira_ticket)
+                    jira_tickets_for_playbooks[playbook_id] = jira_ticket
                     logging.debug(f"Skipped updating Jira issue for resolved test playbook:{playbook_id}")
                     continue
                 junit_file_name = get_attachment_file_name(playbook_id, options.build_number)
                 jira_ticket = create_jira_issue(jira_server, jira_ticket, xml, playbook_id, options.build_number, tabulate_data,
                                                 options.max_days_to_reopen, now, junit_file_name, total_errors > 0)
-                playbook_to_jira_mapping[playbook_id] = jira_ticket_to_json_data(jira_ticket)
+                jira_tickets_for_playbooks[playbook_id] = jira_ticket
             else:
                 logging.debug(f"Skipped creating Jira issue for successful test playbook:{playbook_id}")
 
+        write_test_playbook_to_jira_mapping(artifacts_path, jira_tickets_for_playbooks)
```

Comment:
moved the file write to be the last step, so it will only write it if all the previous code succeeded

Reply:
Wouldn't we benefit if the file will be partly written, if there's some exception?

Reply:
it's partially written in the results step, it's overridden by this step. that's why i'm ok with it here.